usr/src/uts/common/fs/zfs/dmu.c

```diff
@@ -1091,7 +1091,9 @@ dmu_object_remap_indirects(objset_t *os, uint64_t object,
 		return (err);
 	}
 
-	if (dn->dn_nlevels <= 1) {
+	rw_enter(&dn->dn_struct_rwlock, RW_WRITER);
+	if (dn->dn_phys->dn_nlevels <= 1) {
+		rw_exit(&dn->dn_struct_rwlock);
```

Comment:
Shouldn't there be another rw_exit at the end of this if?

Reply:
This was mismerged, thanks for catching it Andy.

@brad-lewis See DLPX-56613 and DLPX-52564.